csv2wiki/__init__.py

```diff
@@ -838,8 +856,11 @@ def _make_page(self, row):
 
         # Crawl down the page skel, appending page content as needed.
         page_text = ""
+
+        # We wikiize the whole row once so sections can reuse
+        wikiized_row = [self._wikiize_cell(page_title, row, col_idx) for col_idx in range(len(row))]
```

Comment:
Hmmmrm.  The order of the logic here worries me.  You are wikiizing the row *before* doing the {N} substitution.  But shouldn't wikiizing happen post-substitution, so that any content that got brought in via the {N}-substitution gets wikiized?  That was how it was in the old code (via the line `cell = self._format_cell(row[col])`, which was called inside `_do_skel()`).

Note that this is complicated a bit by the question of how we handle categories.  The old code had (IMHO) a small problem, which was that if we encountered a column more than once (i.e., the sec_map happened to specify that column multiple times, for whatever reason), *and* that column happened to be the one named in cat_col, then we would do categories-expansion multiple times.  I think you are right that categories-expansion should be done only once, independently of how many times the cat_col is used in an {N} expression, so maybe *that* processing needs to happen first, but it can be independent of the wikiization of the rows.

Then on *top* of that complication is another complication: should we do categories-expansion at all if the cat_col is never even named in any {N} in a text field?  I mean, before we ever do any text processing, we could just jump to the cat_col in the current row and do categories-expansion there (and emit the result now or later, whatever).  But is that the right thing to do if it turns out that that column is never actually used?  (And it's not like we have an easy test to see whether a given column is ever used... we're handing the row to the Python string formatting parser, and it's taking care of the {N} processing for us; we never actually see which N's were used and which weren't.)